README.md

```diff
@@ -1,7 +1,100 @@
-Generate a compatibility matrix for plugins against Jenkins core.
+Plugin Compatibility Tester (PCT)
+------
+
+Generates a compatibility matrix for plugins against Jenkins core.
 
 See https://wiki.jenkins-ci.org/display/JENKINS/Plugin+Compatibility+Tester for background.
 
+## Running PCT 
+
+### Running PCT in Docker
+
+It is recommended to run PCT in the prepared Docker image.
+You can find it [here](hub.docker.com/r/jenkins/pct/).
+
+#### Examples
+
+This command will clone of a repository and run PCT against the latest Jenkins Core:
+
+```shell
+docker run --rm -v maven-repo:/root/.m2 -v $(pwd)/out:/pct/out -e ARTIFACT_ID=ssh-slaves -e VERSION=ssh-slaves-1.24 jenkins/pct
```

Comment:
You _can_ use volumes to cache artifacts between PCT runs, though this is wasteful as you do not share artifacts with your own `~/.m2/repository/`. [With the `maven` image](https://github.com/carlossg/docker-maven/blob/93d297ed2fc952af8c3638eae78c3d5e7526033f/jdk-8/mvn-entrypoint.sh), you can just use

```bash
-v ~/.m2:/var/maven/.m2 -u $(id -u):$(id -g) -e MAVEN_CONFIG=/var/maven/.m2
```

so that the container simply reuses your own local repository, reading and writing as your own UID.

Reply:
I can change an example if you prefer. But it does not really matter from where the volume comes. In my example you can use `-v ~/.m2:/root/.m2` (assuming the container runs as root, of course)

Reply:
@oleg-nenashev @jglick I have found problems with -u when you run inside docker containers (for example on CI with docker agents) as typically you are root inside the container

Reply:
Yes, this image will not work if you run it as non-root. It may be a problem for Docker Pipeline plugin which passes a custom user by default IIRC. I could map the image to `/pct/.m2` and then make tweaks to make Maven to pick the images from there. WDYT?

Reply:
During the work on `runPCT()` (https://github.com/jenkins-infra/pipeline-library/pull/35) we discovered that mapping volumes is not a good option for production setup, because only one container is mapped with write permissions. There I fell back to external staging repo.

So there is no demand in it right now.